openage/convert/main.py

```diff
@@ -170,35 +170,36 @@ def conversion_required(asset_dir, args):
     Sets options in args according to what sorts of conversion are required.
     """
     try:
-        with asset_dir['converted', ASSET_VERSION_FILENAME].open() as fileobj:
+        with asset_dir['converted',
+                       changelog.ASSET_VERSION_FILENAME].open() as fileobj:
             asset_version = fileobj.read().strip()
     except FileNotFoundError:
         # assets have not been converted yet
         info("No converted assets have been found")
         return True
 
+    # TODO: datapack parsing
+
     try:
         asset_version = int(asset_version)
     except ValueError:
         info("Converted assets have improper format; expected integer version")
         return True
 
-    # this is the place where we can do sophisticated checks whether
-    # the assets are outdated, and select which parts require reconversion.
-    if asset_version >= 1:
+    changes = changelog.changes(asset_version)
+
+    if not changes:
         dbg("Converted assets are up to date")
         return False
 
-    info("Converted assets outdated: Version " + str(asset_version))
-
-    if asset_version == 0:
-        info("Re-converting only sounds")
-        args.no_metadata = True
-        args.no_graphics = True
-        return True
+    else:
+        info("Converted assets outdated: Version {}".format(asset_version))
+        info("Reconverting {}".format(",".join(sorted(changes))))
```

Comment:
nah, that would require accessing `changelog` "internals", and its more informative to the user if the program says what it means by everything.